### PR TITLE
Settings: Improve Site Kit detection logic

### LINKED ISSUE #5877: Settings: only disable GA input field if Site Kit Analytics module is active
## Feature Description

Right now, the input field to enter the Google Analytics ID is disabled even when Site Kit is installed, but not actually active. It makes more sense to do this when Site Kit is active, and its Analytics module is active as well.

Otherwise this is confusing for users ([case in point](https://wordpress.org/support/topic/story-not-displaying-properly/#post-13894326)).

Code in question:

https://github.com/google/web-stories-wp/blob/ed95fb387de1921cd6906980f3c9dd5b7dc9a850/assets/src/dashboard/app/views/editorSettings/googleAnalytics/index.js#L178

Should check for `siteKitCapabilities.analyticsModuleActive` instead of `siteKitCapabilities.siteKitInstalled`.

The helper text can still be displayed when Site Kit is installed, as we then simply point the user to activate the plugin. 

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by samwhale**
## Summary

The google analytics input is only disabled when the site analytics module is installed and active. Otherwise, the input is enabled.

## Relevant Technical Choices

- the input will only be disabled when the site kit plugin is installed and activated.

## To-do

none

## User-facing changes

- the input is disabled when the site kit plugin is installed and activated. It used to be disabled if the site kit plugin was just installed.

## Testing Instructions

- start out with `Site Kit` plugin not installed
- go to stories dashboard settings. Should see text prompting you to install the plugin.
- add `Site Kit` plugin from google to wordpress
- go to stories dashboard settings. Should see text prompting you to activate the plugin
- You can't activate the plugin in a staging environment, but if you did you'd see the 3rd screenshot below:

|||
|--|--|
|No Site Kit|image|
|Site Kit Installed|image|
|Site Kit Activated (input is disabled)|image|

---

Fixes #5877

### CODE REVIEW

**Comment by samwhale** [comment]
Haven't been able to get `analyticsModuleActive` to be `true` when running this locally. @swissspidy a couple questions for you:

1. How do you get this plugin to change its `analyticsModuleActive` flag to `true`? 
2. I'm unsure of how to get the mock to set this flag to true. Could you point me in the right direction? php is not my forté 😕

**Comment by swissspidy** [comment]
> Haven't been able to get `analyticsModuleActive` to be `true` when running this locally. @swissspidy a couple questions for you:
> 
> 1. How do you get this plugin to change its `analyticsModuleActive` flag to `true`?
> 2. I'm unsure of how to get the mock to set this flag to true. Could you point me in the right direction? php is not my forté 😕

Very good questions!

I just looked through the code and there seems to be something very strange going on, as if we messed up some merge at some point. Let me take a closer look...

**Comment by swissspidy** [comment]
@samwhale So basically that flag should only be true when you activate the analytics module. Locally that is not possible. So that mock plugin is an alternative. With it, it will always be true.

I just pushed some code that should fix this logic.

**Comment by spacedmonkey** [comment]
Screenshot 2021-01-15 at 12 59 29
Two things about this message. First, when clicking on it doesn't take me to the screen to activate it, takes me to an extrenal link, not what I was expecting at all. Second, there is no check anymore to see if the current user can activate plugins.

File: includes/Dashboard.php [hunk 1]
```diff
@@ -347,38 +347,31 @@ public function get_dashboard_settings() {
...
 					'canManageSettings'   => current_user_can( 'manage_options' ),
 					'canUploadFiles'      => current_user_can( 'upload_files' ),
 					'canReadPrivatePosts' => $can_read_private_posts,
 				],
-				'siteKitCapabilities' => [
-					'siteKitInstalled'      => $this->is_site_kit_plugin_installed(),
-					'siteKitActive'         => defined( 'GOOGLESITEKIT_VERSION' ),
-					'analyticsModuleActive' => false, // TODO: copy the logic from Analytics.php we need to share somehow.
-					'canInstallPlugins'     => current_user_can( 'install_plugins' ),
-					'canActivatePlugins'    => current_user_can( 'activate_plugin', 'google-site-kit/google-site-kit.php' ),
```
Comment (spacedmonkey) [inline includes/Dashboard.php:379]: Why were the capabilities check removed here? This seems like a regression over an improvement.

File: assets/src/dashboard/karma/fixture.js [hunk 2]
```diff
@@ -40,12 +40,11 @@ const defaultConfig = {
...
-    analyticsModuleActive: false,
-    canActivatePlugins: true,
-    canInstallPlugins: true,
-    siteKitActive: false,
-    siteKitInstalled: false,
+  siteKitStatus: {
+    installed: false,
+    active: false,
+    analyticActive: false,
+    link: 'https://example.com/wp-admin/plugins.php'
```
Comment (spacedmonkey) [inline assets/src/dashboard/karma/fixture.js]: Seems valid.

File: includes/Dashboard.php (hunk 1 above)
Comment (swissspidy) [reply includes/Dashboard.php:379]: It's unneeded code.

**Comment by swissspidy** [comment]
> Screenshot 2021-01-15 at 12 59 29
> 
> Two things about this message. First, when clicking on it doesn't take me to the screen to activate it, takes me to an extrenal link, not what I was expecting at all.

If Site Kit is installed, you'll be sent to the plugin activation screen. If not, you'll be sent to WordPress.org. This has been the case before, and this is the case now. No change in behavior there.

>  Second, there is no check anymore to see if the current user can activate plugins.

The check is only needed on the server side in the `Site_Kit` class.

File: assets/src/dashboard/app/views/editorSettings/googleAnalytics/index.js [hunk 3]
```diff
@@ -204,12 +181,11 @@ function GoogleAnalyticsSettings({
...
-    canActivatePlugins: PropTypes.bool,
-    canInstallPlugins: PropTypes.bool,
-    siteKitActive: PropTypes.bool,
-    siteKitInstalled: PropTypes.bool,
+  siteKitStatus: PropTypes.shape({
+    installed: PropTypes.bool,
+    active: PropTypes.bool,
+    analyticsActive: PropTypes.bool,
+    link: PropTypes.string,
   }),
```
Comment (BrittanyIRL) [inline assets/src/dashboard/app/views/editorSettings/googleAnalytics/index.js:189]: big fan of this update to the structure of these props.

**Review by BrittanyIRL — APPROVED** [review]
This looks good to me. 

I love the naming updates and the simplified logic for what text to display. Thank you for the storybook updates to keep that current too. 

Thank you for helping with this, @swissspidy !!

**Comment by samwhale** [comment]
thank you @swissspidy for all the help on this pr 🙇

File: assets/src/dashboard/app/views/editorSettings/googleAnalytics/index.js (hunk 3 above)
Comment (embarks) [reply assets/src/dashboard/app/views/editorSettings/googleAnalytics/index.js:189]: same, a lot better 👍

**Comment by embarks** [comment]
>start out with Site Kit plugin not installed
>go to stories dashboard settings. Should see text prompting you to install the plugin.
>add Site Kit plugin from google to wordpress
>go to stories dashboard settings. Should see text prompting you to activate the plugin

Works as you say here for me. Great job. 👍

**Review by embarks — APPROVED** [review]

**Pull request merged**